python/paddle/fluid/framework.py

```diff
@@ -543,6 +545,16 @@ def __init__(self,
             self._stop_gradient = stop_gradient
             self.is_data = is_data
 
+    def detach(self):
+        new_var = self._cloneVar()
```

Comment:
use `in_dygraph_mode()` to make sure this will not be used in static graph mode, and throw error when user use it. What's more, add Doc
for this API just like other apis

Reply:
done